### PR TITLE
PMM-9150: override all props from user provided config

### PR DESCRIPTION
**Pull request by ritbl**

### CODE REVIEW

File: ci.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 logging.basicConfig(stream=sys.stdout, format='[%(levelname)s] %(asctime)s: %(message)s', level=logging.INFO)
 
 YAML_CONFIG = 'ci-default.yml'
-YAML_CONFIG_CUSTOM = 'ci.yml'
+YAML_CONFIG_OVERRIDE = 'ci.yml'
```
Comment (ritbl) [inline ci.py:21]: It is not a custom config per se, but more an override.

File: ci.py [hunk 2]
```diff
@@ -28,35 +28,35 @@ class Builder():
...
             return yaml.load(f, Loader=yaml.FullLoader)
 
     def write_custom_config(self, config):
-        with open(YAML_CONFIG_CUSTOM, 'w') as f:
+        with open(YAML_CONFIG_OVERRIDE, 'w') as f:
             yaml.dump(config, f, sort_keys=False)
 
     def merge_configs(self):
-        if self.custom_config is not None:
-            # Yep we have high complexity here but list is short
```
Comment (ritbl) [inline ci.py:51]: I think this comment make no sense, creating map could not be beneficial as well. It needs to be tested.
Plus, it is Python. It needs to be clear, not fast ;)

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-2134-554c7ae
client docker - perconalab/pmm-client-fb:PR-2134-554c7ae
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-2134-554c7ae.tar.gz
Create Staging Instance: https://pmm.cd.percona.com/job/aws-staging-start/parambuild/?DOCKER_VERSION=perconalab/pmm-server-fb:PR-2134-554c7ae&CLIENT_VERSION=https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-2134-554c7ae.tar.gz

**Review by nikita-b — APPROVED** [review]
LGTM

**Comment by JNKPercona** [comment]
API tests have failed, Please check: API: https://pmm.cd.percona.com/job/pmm2-api-tests/8489/

**Pull request merged**